addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/SerialThingHandler.java

```diff
@@ -94,33 +97,15 @@ public void writeString(String msg) {
     }
 
     @Override
-    public void serialEvent(SerialPortEvent event) {
-
-        switch (event.getEventType()) {
-            case SerialPortEvent.BI:
-            case SerialPortEvent.OE:
-            case SerialPortEvent.FE:
-            case SerialPortEvent.PE:
-            case SerialPortEvent.CD:
-            case SerialPortEvent.CTS:
-            case SerialPortEvent.DSR:
-            case SerialPortEvent.RI:
-            case SerialPortEvent.OUTPUT_BUFFER_EMPTY:
-                break;
-            case SerialPortEvent.DATA_AVAILABLE:
-                try {
-                    BufferedReader br = new BufferedReader(new InputStreamReader(inputStream), bufferSize);
-                    while (br.ready()) {
-                        String line = br.readLine();
-                        logger.debug("Receiving '{}' on '{}'", line, getConfig().get(PORT));
-                        onDataReceived(line);
-                    }
-                } catch (IOException e) {
-                    String port = (String) getConfig().get(PORT);
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            "Error receiving data on serial port " + port + " : " + e.getMessage());
-                }
-                break;
+    public void serialEvent(SerialPortEvent arg0) {
+        try {
+            /*
+             * See more details from
+             * https://github.com/NeuronRobotics/nrjavaserial/issues/22
```

Comment:
As you can see from my comment, this link didn't help me to understand the issue - and a `Thread.sleep(Long.MAX_VALUE);` will ring red alarm bells for anyone looking at that code.
Could you leave a comment in the sources here, why this workaround is still needed despite the fact that the referenced issue is closed and marked as solved?